backend/src/users/user.service.ts

```diff
@@ -22,14 +22,33 @@ export class UsersService {
         return this.userModel.find().exec();
     }
 
-    async findOne(@Param(':id') id: number | string) {
+    async findById(@Param(':id') id: number | string) {
         const users = await this.userModel.find().exec()
```

Comment:
Instead of querying for every user and then filtering by comparing ids, just user native `findById` method that all models have.